### PR TITLE
packages/umoci

### PR DESCRIPTION
**Pull request by DarylGrunau**
Support to build/install umoci, an Open Container image manipulation tool

### CODE REVIEW

**Comment by adamjstewart** [comment]
Can you explain the reasoning behind `expand=False` and the manual tarball extraction?

**Comment by DarylGrunau** [comment]
I left `expand=False` because the package sources really, really wanted to live in a particular subdirectory (the one that gets manufactured in the spec) in order to be properly built by go.

File: var/spack/repos/builtin/packages/umoci/package.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Umoci(MakefilePackage):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/umoci/package.py]: Won't the build phase crash without a Makefile?

Comment (DarylGrunau) [reply var/spack/repos/builtin/packages/umoci/package.py]: Sorry for my fumbling around here - I think I finally got it right.  The new build phase creates the directory infrastructure and untars the src, which contains the Makefile.  Re: your earlier questions about leaving `expand=False` I started going down the path of editing the Makefile and setting the `PROJECT = .` (go apparently does not like FQPN's).  This led to a nightmare of fetching additional go packages off of github and placing them in proper subdirs using spack resources.  I eventually hit a roadblock when some subpackage required umoci to also live in one of these subdirs, so I ended up just giving in ...

File: var/spack/repos/builtin/packages/umoci/package.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    depends_on('go')
+
+    def build(self, spec, prefix):
+        mkdirp(join_path(self.stage.source_path,
+               'src', 'github.com', 'openSUSE', 'umoci'))
+
+        with working_dir(join_path('src', 'github.com', 'openSUSE', 'umoci')):
+            tar = which('tar')
+            tar('zxf', self.stage.archive_file, '--strip-components=1')
```
[the remark this answers is not in the document]
Comment (DarylGrunau) [reply var/spack/repos/builtin/packages/umoci/package.py]: I took a gander at the RPM spec file for this package and mimicked its behavior to relax the `expand=False` requirement

**Pull request merged**